src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -722,7 +722,7 @@ function ($input) {
                     } else {
                         throw new LogicException("Invalid string escape: " . $text);
                     }
-                } elseif ($id === T_CLASS_C) {
+                } elseif ($id === T_CLASS_C || $id === T_TRAIT_C) {
```

Comment:
Above we're setting `$id` to `T_TRAIT` if that's available, but it looks like we're only ever doing anything with it if it's set to `T_TRAIT_C`.... Should we just always set it to `T_TRAIT` above for consistency, so we make sure we always handle it here?

Reply:
I don't think so

`$id` gets get to `T_TRAIT` via 

```
$tokens = token_get_all("<?php\n" . $content);
...
  foreach ($tokens as $token) {
    ...
    list($id, $text) = $token;
```

We then change it to T_TRAIT_C via
`$id = $id === T_TRAIT ? T_TRAIT_C : T_CLASS_C;`

For this PR I've simply adapted the existing code that handles `__CLASS__` tokens inside a `class` block to handle `__TRAIT__` inside a `trait` block.

Reply:
Ahhh okay I misread that `$id` assigment line. No worries then.